### ISSUE TITLE
Trying to access 'resultsHandle' variable before initialization.

### ISSUE THREAD

**Issue by johannel00** [matteodem/meteor-easy-search#655]
Trying to access this variable before initialization causes an error.

https://github.com/matteodem/meteor-easy-search/blob/5fe52ad3c1064851653741f1356759be4dc4b0ca/packages/easysearch_core/lib/core/search-collection.js#L203

**Comment by matteodem** [comment]
Can you give me more context on what it is that you want to achieve?

**Comment by johannel00** [comment]
I was trying to create an index in one of my projects. Unfortunately I can't remember exactly what I was doing at that time, but this line kept on giving me an error and had to stop. I guess this issue can be closed now. I'll leave another comment if I stumble across the error again.

**Comment by johannel00** [comment]
Isn't the 'resultsHandle' variable subjected to a temporal dead zone when you're expecting to stop the cursor observer on line 203 when you're actually initializing the variable on line 216 with `let resultsHandle = cursor.mongoCursor.observe({ ... });` ?

**Comment by TechplexEngineer** [comment]
I'm seeing this stacktrace when running my app on galaxy.
```
Exception in onStop callback: ReferenceError: Cannot access 'resultsHandle' before initialization
 Subscription.<anonymous> (packages/easysearch:core/lib/core/search-collection.js:203:9)
 runWithEnvironment (packages/meteor.js:1286:24)
 packages/meteor.js:1299:14
 packages/ddp-server/livedata_server.js:1158:7
 Array.forEach (<anonymous>)
 Function._.each._.forEach (packages/underscore.js:139:11)
 Subscription._callStopCallbacks (packages/ddp-server/livedata_server.js:1157:7)
 Subscription._deactivate (packages/ddp-server/livedata_server.js:1147:10)
 Subscription.subscriptionProto._deactivate (packages/mdg:meteor-apm-agent/lib/hijack/wrap_subscription.js:75:24)
 packages/ddp-server/livedata_server.js:896:11
 Map.forEach (<anonymous>)
 Session._deactivateAllSubscriptions (packages/ddp-server/livedata_server.js:895:21)
 packages/ddp-server/livedata_server.js:450:12
 Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
 packages/meteor.js:550:25
 runWithEnvironment (packages/meteor.js:1286:24)
```

I'm going to try moving this block of code:
https://github.com/matteodem/meteor-easy-search/blob/5fe52ad3c1064851653741f1356759be4dc4b0ca/packages/easysearch_core/lib/core/search-collection.js#L201-L204

Below this block:
https://github.com/matteodem/meteor-easy-search/blob/5fe52ad3c1064851653741f1356759be4dc4b0ca/packages/easysearch_core/lib/core/search-collection.js#L216-L302

I'm guessing this has to do with newer versions of node being more strict about the scope of let variables.

My meteor app is 1.11.1
which uses node 12.18.4

**Comment by johannel00** [comment]
@TechplexEngineer Let me know if and where I can help. Are you going to create a PR for this?

**Comment by matteodem** [comment]
I'm sorry for being inactive on this project. If anyone creates a PR i'm happy to check it out.

**Comment by TechplexEngineer** [comment]
Still trying to figure out a solid reproduction. I only see this every other week or so. I'd like to see more evidence that my fix works and do any break anything before submitting pull request.

@johannel00 have you been able to reproduce this?

**Comment by johannel00** [comment]
@TechplexEngineer Haven't been able to reproduce this yet. I have since moved to other projects than the one I had this error with.

> I'm guessing this has to do with newer versions of node being more strict about the scope of let variables.

I think you're correct by assuming this. The machine was running the project on had a newer version of node than the v12.16.1 running on mine.

**Comment by matteodem** [comment]
I guess i can provide a fix where the variable is defined before it tries to access it. I'll note it and do it in the next few days. A PR for that is also very easy (if you are up for that). 😄

**Comment by TechplexEngineer** [comment]
PR #656 has the change I deployed to prod. It didn't blow up. I'll happily report back when I have more data. But initial indications look good

**Comment by johannel00** [comment]
> PR #656 has the change I deployed to prod. It didn't blow up. I'll happily report back when I have more data. But initial indications look good

Thanks @TechplexEngineer

**Comment by matteodem** [comment]
Just released the new version `2.2.3` 🎉

**Comment by TechplexEngineer** [comment]
Thanks!
